lint/dupArg_checker.go

```diff
@@ -0,0 +1,89 @@
+package lint
+
+import (
+	"go/ast"
+
+	"github.com/go-toolsmith/astequal"
+)
+
+func init() {
+	addChecker(&dupArgChecker{}, attrExperimental)
+}
+
+type dupArgChecker struct {
+	checkerBase
+
+	matchers map[string]func(*ast.CallExpr) bool
+}
+
+func (c *dupArgChecker) InitDocumentation(d *Documentation) {
+	d.Summary = "Detects suspicious duplicated arguments"
+	d.Before = `copy(dst, dst)`
+	d.After = `copy(dst, src)`
+}
+
+func (c *dupArgChecker) Init() {
+	newMatcherFunc := func(pairs [][2]int) func(*ast.CallExpr) bool {
```

Comment:
As I understood `pairs` contains a pair of argument indexes, right?
Can we make name more descriptive? or add a comment for a `newMatcherFunc`?

Reply:
How about `indexes` and adding a comment for `newMatcherFunc`?

Reply:
Good enough :)

Reply:
`argIndexPairs`? Give a hint. :)
Different algorithm is a hint too.